crates/iceberg/src/table.rs

```diff
@@ -16,28 +16,156 @@
 // under the License.
 
 //! Table API for Apache Iceberg
-use typed_builder::TypedBuilder;
+
+use std::sync::Arc;
 
 use crate::arrow::ArrowReaderBuilder;
+use crate::io::object_cache::ObjectCache;
 use crate::io::FileIO;
 use crate::scan::TableScanBuilder;
 use crate::spec::{TableMetadata, TableMetadataRef};
-use crate::{Result, TableIdent};
+use crate::{Error, ErrorKind, Result, TableIdent};
+
+/// Builder to create table scan.
+pub struct TableBuilder {
```

Comment:
We frequently used `typed_builder`. Does it work well in this situation too?

Reply:
Unfortunately I could not get it to work with `typed_builder`, despite spending a long time trying. I could not configure `typed_builder` to be able to initialise the object cache with either an optional capacity, default capacity or disabled.